### PR TITLE
Proxy from nginx to snowpack dev server only

### PR DESCRIPTION
**Pull request by danr**
Changing to this setup by @pontus from before because the snowpack dev server doesn't proxy `UPDATE` properly.

Dev server is now at localhost:8443 and localhost:8080 again.

### CODE REVIEW

File: system/herdbook.devel.nginx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        location /api/ {
+                # There were issues using the wsgi transport for uwsgi,
+                # use its built in http server.
+                proxy_pass http://127.0.0.1:9090/api/;
+        }
+
+        location / {
+               proxy_pass http://herdbook-frontend-devel:4242;
+               proxy_http_version 1.1;
+               proxy_set_header Upgrade $http_upgrade;
```
Comment (danr) [inline system/herdbook.devel.nginx:18]: Do we need these three `proxy_set_header`? @pontus

Comment (pontus) [reply system/herdbook.devel.nginx:18]: Does snowpack use a wss to transport state information back and forth? In that case yes (at least the Upgrade ones).

**Review by pontus — COMMENTED** [review]
If onw wants to serve static content first properly, we probably need to do something along the lines of https://github.com/NBISweden/theherdbook/blob/766ecdcfcc499845a4259bd286632148c40da0ff/system/herdbook.devel.nginx, alternatively serve at least .well-known statically (or decide that isn't an important use case for the devel, mostly depends on whatever one wants to be able to get ACME certs [e.g. letsencrypt] through what's offered, can likely be achieved by pointing  to a statically served area within what snowpack sees or using another method)
.

**Review by pontus — APPROVED** [review]

**Comment by danr** [comment]
> Does snowpack use a wss to transport state information back and forth? In that case yes (at least the Upgrade ones).

OK! Yes, it uses a websocket. (It only pushes information to the browser though)

> If onw wants to serve static content first properly, we probably need to do something along the lines of https://github.com/NBISweden/theherdbook/blob/766ecdcfcc499845a4259bd286632148c40da0ff/system/herdbook.devel.nginx, alternatively serve at least .well-known statically (or decide that isn't an important use case for the devel, mostly depends on whatever one wants to be able to get ACME certs [e.g. letsencrypt] through what's offered, can likely be achieved by pointing to a statically served area within what snowpack sees or using another method)

I don't understand at all. What part of the linked file are you refrring to? Then what is _serve well-known statically_? Also what does the rest of the paragraph mean?

**Comment by pontus** [comment]
Sorry, lines 23-25 (to replace 15-21 in https://github.com/NBISweden/theherdbook/blob/b558e7ec567c140665332882380059db4cf7e2f5/system/herdbook.devel.nginx, but with the header setting added) and 33-36.

And the use case that is supported by that is serving the same static content in development as is served in production (config/updates is brought in).

And one use case for that is getting certificates through lets encrypt with the "host a file" method, which means your client puts a file that is served as `.well-known/something`.

**Review by norling — APPROVED** [review]
It Works™!

**Comment by danr** [comment]
Now I understand much better, thanks @pontus. Is it important to run let's encrypt while developing though? Or would that be to test certbot setup on the dev server?

**Comment by pontus** [comment]
Probably not an important use case. My original impression when designing was to support frequent refreshes (git pull) of a user visible site.

In practice, I think PRs have been rather large and frequently done changes requiring full rebuilds anyway, so having the user visible site "static" is not that big of a drawback.

**Comment by danr** [comment]
OK let's not bother with that until it becomes important then imo

**Pull request merged**